examples/pytorch_distributed_simple.py

```diff
@@ -0,0 +1,197 @@
+        p = trial.suggest_float("dropout_l{}".format(i), 0.2, 0.5)
+        layers.append(nn.Dropout(p))
+
+        in_features = out_features
+    layers.append(nn.Linear(in_features, CLASSES))
+    layers.append(nn.LogSoftmax(dim=1))
+
+    return nn.Sequential(*layers)
+
+
+def get_mnist():
+    # Load MNIST dataset.
+    train_dataset = datasets.MNIST(DIR, train=True, transform=transforms.ToTensor())
+    train_dataset = torch.utils.data.Subset(train_dataset, indices=range(N_TRAIN_EXAMPLES))
+    train_sampler = torch.utils.data.distributed.DistributedSampler(dataset=train_dataset)
+
+    valid_dataset = datasets.MNIST(DIR, train=False, transform=transforms.ToTensor())
+    valid_dataset = torch.utils.data.Subset(valid_dataset, indices=range(N_VALID_EXAMPLES))
+    valid_sampler = torch.utils.data.distributed.DistributedSampler(
+        dataset=valid_dataset, shuffle=False
+    )
+
+    train_loader = torch.utils.data.DataLoader(
+        train_dataset,
+        sampler=train_sampler,
+        batch_size=BATCHSIZE,
+        shuffle=False,
+    )
+    valid_loader = torch.utils.data.DataLoader(
+        valid_dataset,
+        sampler=valid_sampler,
+        batch_size=BATCHSIZE,
+        shuffle=False,
+    )
+
+    return train_loader, valid_loader, train_sampler, valid_sampler
+
+
+def objective(single_trial):
+    trial = optuna.integration.TorchDistributedTrial(single_trial)
+
+    # Generate the model.
+    model = DDP(define_model(trial).to(DEVICE))
+
+    # Generate the optimizers.
+    optimizer_name = trial.suggest_categorical("optimizer", ["Adam", "RMSprop", "SGD"])
+    lr = trial.suggest_float("lr", 1e-5, 1e-1, log=True)
+    optimizer = getattr(optim, optimizer_name)(model.parameters(), lr=lr)
+
+    # Get the MNIST dataset.
+    train_loader, valid_loader, train_sampler, valid_sampler = get_mnist()
+
+    accuracy = 0
+    # Training of the model.
+    for epoch in range(EPOCHS):
+        model.train()
+        # Shuffle train dataset.
+        train_sampler.set_epoch(epoch)
+        for data, target in train_loader:
+            data, target = data.view(data.size(0), -1).to(DEVICE), target.to(DEVICE)
+
+            optimizer.zero_grad()
+            output = model(data)
+            loss = F.nll_loss(output, target)
+            loss.backward()
+            optimizer.step()
+
+        # Validation of the model.
+        model.eval()
+        correct = 0
+        with torch.no_grad():
+            for data, target in valid_loader:
+                data, target = data.view(data.size(0), -1).to(DEVICE), target.to(DEVICE)
+                output = model(data)
+                # Get the index of the max log-probability.
+                pred = output.argmax(dim=1, keepdim=True)
+                correct += pred.eq(target.view_as(pred)).sum().item()
+
+        correct_tensor = torch.tensor([correct], dtype=torch.int)
+        dist.all_reduce(correct_tensor)
+        total_correct = correct_tensor.numpy().tolist()[0]
```

Comment:
```suggestion
        total_correct = correct_tensor.item()
```

Reply:
I think that `.item()` is enough to access the value.

Reply:
Thank you for your suggestion. We can apply the same change to `_call_and_communicate` and `_call_and_communicate_obj`, so I'll update them.

Reply:
Updated in https://github.com/optuna/optuna/pull/2303/commits/496a1d0098d223475b288381687c7f2e0b5842eb. According to the [reference of `.tolist()`](https://pytorch.org/docs/stable/tensors.html#torch.Tensor.tolist), data will be moved from GPU to cpu automatically. I removed `to("cpu").numpy()` because it is redundant.

Reply:
Nice!